editor/src/clj/editor/pipeline/fontc.clj

```diff
@@ -720,26 +721,31 @@
-                    {:keys [image-wh glyph-cell-wh ^int glyph-data-size ^int glyph-data-offset]} glyph-extents
-                    ^int image-width (:width image-wh)
-                    ^int image-height (:height image-wh)
-                    row-size (* (int (:width glyph-cell-wh)) channel-count)]
-                (Arrays/fill glyph-data-bank glyph-data-offset (+ glyph-data-offset row-size) (unchecked-byte 0))
-                (doseq [^int y (range image-height)]
-                  (let [y-offset (+ glyph-data-offset
-                                    row-size
-                                    (* y row-size))]
-                    (aset glyph-data-bank y-offset (unchecked-byte 0))
-                    (System/arraycopy field (+ (* y image-width channel-count) 0) glyph-data-bank (+ y-offset channel-count) (* image-width channel-count))
-                    (aset glyph-data-bank (+ (+ y-offset channel-count) (* image-width channel-count)) (unchecked-byte 0))))
-                (let [last-y-offset (- (+ glyph-data-offset glyph-data-size) row-size)]
-                  (Arrays/fill glyph-data-bank last-y-offset (+ last-y-offset row-size) (unchecked-byte 0)))))
-            (sequence positive-glyph-extent-pairs-xf
-                      semi-glyphs
-                      glyph-extents)))
+    (dorun
+      (pmap
+        (fn [batch]
+          (run! (fn [[semi-glyph glyph-extents]]
+                  (let [{:keys [^bytes field]} (draw-ttf-distance-field semi-glyph padding channel-count outline-width sdf-outline sdf-spread sdf-shadow-spread edge shadow-blur shadow-alpha)
+                        {:keys [image-wh glyph-cell-wh ^int glyph-data-size ^int glyph-data-offset]} glyph-extents
+                        ^int image-width (:width image-wh)
+                        ^int image-height (:height image-wh)
+                        row-size (* (int (:width glyph-cell-wh)) channel-count)]
+                    (Arrays/fill glyph-data-bank glyph-data-offset (+ glyph-data-offset row-size) (unchecked-byte 0))
+                    (doseq [^int y (range image-height)]
+                      (let [y-offset (+ glyph-data-offset
+                                        row-size
+                                        (* y row-size))]
+                        (aset glyph-data-bank y-offset (unchecked-byte 0))
+                        (System/arraycopy field (+ (* y image-width channel-count) 0) glyph-data-bank (+ y-offset channel-count) (* image-width channel-count))
+                        (aset glyph-data-bank (+ (+ y-offset channel-count) (* image-width channel-count)) (unchecked-byte 0))))
+                    (let [last-y-offset (- (+ glyph-data-offset glyph-data-size) row-size)]
+                      (Arrays/fill glyph-data-bank last-y-offset (+ last-y-offset row-size) (unchecked-byte 0)))))
+                batch))
+        (sequence (comp
+                    positive-glyph-extent-pairs-xf
+                    (partition-all 100))
```

Comment:
Batching improved font compilation speed from 3s to 2s. Also, I did replaced `doall` with `dorun` so we don't keep the seq in memory while performing the drawing. Not that it matters now with batching...